Apps.Okapi/Api/OkapiClient.cs

```diff
@@ -16,25 +16,42 @@ public async Task<T> ExecuteWithXml<T>(string endpoint, Method method, object? b
         var serializer = new XmlSerializer(typeof(T));
         using var reader = new StringReader(response.Content);
         var result = (T)serializer.Deserialize(reader)!;
-        
+
         return result;
     }
-    
+
     public async Task<RestResponse> Execute(string endpoint, Method method, object? bodyObj,
         AuthenticationCredentialsProvider[] creds)
     {
         var baseUrl = creds.Get(CredsNames.Url).Value
             .TrimEnd('/');
-        
+
+        var request = new OkapiRequest(new()
+        {
+            Url = baseUrl + endpoint,
+            Method = method
+        });
+
+        return await ExecuteRequest(request);
+    }
+
+    public async Task<RestResponse> UploadFile(string endpoint, Method method, FileParameter fileParam,
+        AuthenticationCredentialsProvider[] creds)
+    {
+        var baseUrl = creds.Get(CredsNames.Url).Value
+            .TrimEnd('/');
+
         var request = new OkapiRequest(new()
         {
             Url = baseUrl + endpoint,
             Method = method
         });
 
+        request.AddFile(fileParam.Name, () => fileParam.GetFile(), fileParam.FileName, fileParam.ContentType);
```

Comment:
For unknown reason, document says it's possible to pass a `FileParameter` to `request.AddFile`, but this overload is not actually implemented 🤷‍♂️

Reply:
I've covered it with test and just doubled checked in a fly, it succeeded:
https://us-1.blackbird.io/flights/681a758c8b773a2feeca729f

Do you have any specific edge case in mind where it could fail?